rules/module_extensions.bzl

```diff
@@ -0,0 +1,44 @@
+"""Definitions for bzlmod module extensions."""
+
+load(
+    "//rules:repositories.bzl",
+    "rules_ios_dependencies",
+    "rules_ios_dev_dependencies",
+)
+load(
+    "//tools/toolchains/xcode_configure:xcode_configure.bzl",
+    _xcode_configure = "xcode_configure",
+)
+
+def _non_module_deps_impl(_):
+    rules_ios_dependencies(
+        load_xchammer_dependencies = True,
+        load_bzlmod_dependencies = False,
+    )
+
+non_module_deps = module_extension(implementation = _non_module_deps_impl)
```

Comment:
This loads things not supported in the workspace (i.e. the repository rules) we had in WORKSPACE previously